tools/src/OpenSearchHttpClient.ts

```diff
@@ -72,26 +115,58 @@ export interface OpenSearchInfo {
 export class OpenSearchHttpClient {
   private readonly _axios: AxiosInstance
   private readonly _opts?: OpenSearchHttpClientOptions
+  private readonly _logger?: Logger
```

Comment:
I prefer having a default null-logger than having to typecheck if `_logger` is null everywhere it is used.